samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -512,20 +518,30 @@ object SamzaContainer extends Logging {
 
     val timerExecutor = Executors.newSingleThreadScheduledExecutor
 
-    // We create a map of store SystemName to its respective SystemConsumer
-    val storeSystemConsumers: Map[String, SystemConsumer] = changeLogSystemStreams.mapValues {
-      case (changeLogSystemStream) => (changeLogSystemStream.getSystem)
-    }.values.toSet.map {
-      systemName: String =>
-        (systemName, systemFactories
-          .getOrElse(systemName,
-            throw new SamzaException("Changelog system %s exist in the config." format (systemName)))
-          .getConsumer(systemName, config, samzaContainerMetrics.registry))
-    }.toMap
+    var taskStorageManagers : Map[TaskName, TaskStorageManager] = Map()
 
-    info("Created store system consumers: %s" format storeSystemConsumers)
+    val taskInstanceMetrics: Map[TaskName, TaskInstanceMetrics] = containerModel.getTasks.values.asScala.map(taskModel => {
+      (taskModel.getTaskName, new TaskInstanceMetrics("TaskName-%s" format taskModel.getTaskName))
+    }).toMap
 
-    var taskStorageManagers : Map[TaskName, TaskStorageManager] = Map()
+    val taskCollectors : Map[TaskName, TaskInstanceCollector] = containerModel.getTasks.values.asScala.map(taskModel => {
+      (taskModel.getTaskName, new TaskInstanceCollector(producerMultiplexer, taskInstanceMetrics.get(taskModel.getTaskName).get))
+    }).toMap
+
+
+    val containerStorageManager = new ContainerStorageManager(containerModel, streamMetadataCache, systemAdmins, changeLogSystemStreams.asJava,
```

Comment:
Why pass `Optional.empty()` for base directory path when instantiating ContainerStorageManager? Instead, should we create the CSM after we resolving the base dir paths?

Reply:
done